pkg/linear-pools/contracts/silo-linear-pool/SiloExchangeRateModel.sol

```diff
@@ -33,16 +33,18 @@ contract SiloExchangeRateModel {
      * @dev This function is similar to _accrueInterest function in the Silo's BaseSilo.sol contract
      * which is used to update state data that is necessary
      */
-    function calculateExchangeValue(IShareToken shareToken) external view returns (uint256) {
-        uint256 rcomp = _getCompoundInterestRate(shareToken.silo(), shareToken.asset());
-        ISilo.AssetStorage memory assetStorage = _getAssetStorage(shareToken.silo(), shareToken.asset());
+    function _calculateExchangeValue(IShareToken shareToken) internal view returns (uint256) {
+        ISilo silo = shareToken.silo();
+        address underlyingAsset = shareToken.asset();
```

Comment:
Same here, I think we should pass in the `underlyingAsset` as an argument because everywhere we call this function, we already have direct access to the `mainToken`. We can take advantage of our existing storage to reduce the number of external calls we make.